core/lib/mcxtx_api.h

```diff
@@ -135,7 +135,15 @@
      * all.  We do not need anything more than word alignment for OP_vldm/OP_vstm,
      * and dr_simd_t has no fields larger than 32 bits, so we have no padding.
      */
+#ifdef AARCH64
+    dr_simd_t simd[MCXT_NUM_SIMD_SVE_SLOTS];
+#else
     dr_simd_t simd[MCXT_NUM_SIMD_SLOTS];
+#endif
+#    ifdef X64 /* 64-bit */
+    dr_simd_t svep[MCXT_NUM_SVEP_SLOTS];
```

Comment:
When appending fields to this public structure we have to carefully only access them if the user-set size field includes them (otherwise we clobber adjacent data and crash the client in unpredictable ways). Things like priv_mcontext_to_dr_mcontext() need to be updated to fill these in if the size field is large enough.

Wow: I just looked at priv_mcontext_to_dr_mcontext() and it doesn't copy *any* SIMD state for aarchxx!  How is that possible?  Are there really no clients setting DR_MC_MULTIMEDIA?!?

Reply:
I see a bunch of tests using DR_MC_ALL: why don't they hit that assert in priv_mcontext_to_dr_mcontext()??

Reply:
> Wow: I just looked at priv_mcontext_to_dr_mcontext() and it doesn't copy any SIMD state for aarchxx! How is that possible? Are there really no clients setting DR_MC_MULTIMEDIA?!?

I can't see any clients which set `DR_MC_MULTIMEDIA`

> I see a bunch of tests using DR_MC_ALL: why don't they hit that assert in priv_mcontext_to_dr_mcontext()??

The `ASSERT_NOT_IMPLEMENTED` is in the else clause of the `DR_MC_ALL` check.
```
    if (TESTALL(DR_MC_ALL, dst->flags) && dst->size == sizeof(dr_mcontext_t)) {
        *(priv_mcontext_t *)(&MCXT_FIRST_REG_FIELD(dst)) = *src;          
    } else {  
. . .
        if (TEST(DR_MC_MULTIMEDIA, dst->flags)) {                         
#ifdef X86
. . .
#else                                                                     
            /* FIXME i#1551: NYI on ARM */                                
            ASSERT_NOT_IMPLEMENTED(false);                                
#endif                                                                    
        }                                                                 
    }                                                                     
    return true;                                                          
} 
```
This functionality hasn't been missed as we have successfully tested this patch on SVE h/w using:
- SPEC2000
- fleetbench
- hashtable
- memcached
- memtier
- Internal proprietary SVE tests suites

They have been run with some sample clients (instrace_simple, memtrace_simple, memval_simple) and drmemtrace but with some memory instrumentation disabled, as highlighted in the `/* TODO i#5844... */` comments.

Reply:
DR_MC_MULTIMEDIA is set by DR_MC_ALL.  OK so the initial check for DR_MC_ALL is what is reached today.  But that single copy is unsafe for these new fields: so that code needs to be updated to handle a client whose size != the new size to avoid breaking binary compatibility.

Reply:
I've replaced the `ASSERT_NOT_IMPLEMENTED()` for AArch64 with copying code for binary compatibility. AFAICT it is simpler than the X86 equivalent for `DR_MC_MULTIMEDIA` because the AArch64 differences are based on the size of `simd` in the case of pre-SVE clients.